### DISCUSSION TITLE
Search definition of function in both source and included files(*.c,*.h) for C-language

### DISCUSSION THREAD

**Discussion by shusterboris** [oracle/opengrok#4352] [Q&A] [answered]
Have a nice day, everyone!
In previous versions, when I searched for a function definition, I got both functions and prototypes (in included files) in the results of searching. In version 1.8.4, which I'm using now, there is only 1 file in the list - the definition of function in c-file.
Is there a way (for example, indexer settings, ctags) to force the system to show header files again?
Best regards!

**Comment by vladak** [comment]
It depends on how you search. When searching for reference (called 'Symbol' in the UI) of a C function, the search should turn functions calls, declarations and definitions.

**Reply by shusterboris** [reply]
Thank you for quick answer, I really impressed! 
But in previous versions (quite old) function's definitions showed up in the results in Definition mode... and my old-school colleagues are worrying... It seems I should search workaround...

**Reply by vladak** [reply, marked as answer]
You probably mean function declarations showed as functions definitions. Ultimately, this boils down to ctags telling the indexer what are the definitions.

**Reply by shusterboris** [reply]
Exactly.

**Comment by shusterboris** [comment]
It seems, to solve the problem, I need to use --c-kinds+px option for
ctags, but I don't understand, how to do that. I would be very appreciated
for help

вт, 16 мая 2023 г., 17:33 Vladimir Kotal ***@***.***>:

**Reply by vladak** [reply]
See https://github.com/oracle/opengrok/wiki/Indexer-configuration#custom-ctags-configuration
